contents/tutorials/web-redact-properties.md

```diff
@@ -0,0 +1,130 @@
+date: 2025-05-29
+showTitle: true
+author:
+  - vincent-ge
+tags:
+  - configuration
+  - data management
+  - product os
+---
+
+When collecting data to better understand your customers, it's important to comply with [privacy regulations](https://posthog.com/docs/privacy). 
+
+Sometimes, this means certain information should **never** make it to PostHog servers. This tutorial shows you how to hide or redact information on the [PostHog JavaScript Web SDK](https://posthog.com/docs/libraries/js), before they're sent to PostHog.
+
+## Redacting information with the Web SDK
+You can selectively remove properties from events before they're sent by initializing PostHog with a [`before_send`](https://posthog.com/docs/libraries/js/features#redacting-information-in-events) hook. PostHog will pass the event object to the `before_send` function, where you can redact any information.
+
+For example, you can create a `beforeSend.js` file and define your redaction logic there.
+
+```js file=beforeSend.js
+// beforeSend.js
+// Define the properties you want to redact
+const redactedProperties = ["url", "href", "pathname", "referrer", "host", "user_agent"];
+
+// Match property names against the defined list
+function filterProperties(value) {
+  return Object.entries(value).reduce((acc, [key, value]) => {
+    if (redactedProperties.some(prop => key.includes(prop))) {
+      acc[key] = null;
+    } else {
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+}
+
+// Export the before send function
+export const beforeSend = (event) => {
+  if (!event) {
+    return null;
+  }
+
+  const redactedProperties = filterProperties(event.properties || {});
+  event.properties = redactedProperties;
+
+  // $set
+  const redactedSet = filterProperties(event.$set || {});
+  event.$set = redactedSet;
+
+  // $set_once
+  const redactedSetOnce = filterProperties(event.$set_once || {});
+  event.$set_once = redactedSetOnce;
+
+  return event;
+};
+```
+
+Then, you can initialize PostHog with your `before_send` function.
+```js
+import { beforeSend } from './beforeSend'
+
+posthog.init('<ph_project_api_key>', {
+  api_host: '<ph_client_api_host>',  // usually 'https://us.i.posthog.com' or 'https://eu.i.posthog.com'
+  capture_pageview: 'history_change',
+  before_send: beforeSend
+})
+```
+
+## Hiding customer IP address
+
+PostHog will use the client IP address found during an event capture request if an `$ip` isn't passed through `event.properties`. To hide your customer's IP address, you can override the `$ip` property with a sample address like `192.0.2.0`.
+
+```js file=beforeSend.js
+export const beforeSend = (event) => {
+  if (!event) {
+    return null;
+  }
+
+  event.properties['$ip'] = '';
+
+  return event;
+};
+```
+
+If you require your customers' IP addresses to be completely obfuscated from PostHog servers, consider [hosting a reverse proxy](/docs/advanced/proxy#deploying-a-reverse-proxy).
```

Comment:
I don't think this is a good suggestion, it will still require them to set up the code to drop the IP address from the event. Toggling IP capture in settings is probably better: [us.posthog.com/project/2/settings/environment#datacapture](https://us.posthog.com/project/2/settings/environment#datacapture)

Reply:
I think this is definitely better. 

I saw a thread where someone wanted IPs to never reach PostHog servers 🤷‍♂️ I _think_ no process/storage does the same. I think maybe that's an edge case and we should give the relevant suggestion case per case.